front-end/src/components/Dashboard/index.js

```diff
@@ -43,21 +50,97 @@ class Dashboard extends React.Component {
   }
 
 
+  hasGH = () =>{
+    if (this.state.github_info === null){
+      return false;
+    } else {
+      return true;
+    }
+  }
+  renderWelcome = () =>{
+    if (this.hasGH()) {
+      return <div> Welcome {this.state.github_info.github_username}</div>;
+    } else{
+      return <div>Welcome {this.state.email}</div>;
+    }
+  }
+  renderGitHubImage = () => {
+    if(this.hasGH()){
+      return <img src={this.state.github_info.profile_image_url} alt='userGHImage' height='150px'/>;
+    } else {
+      return null;
+    }
+  }
+  renderPhone = () => {
+    if (this.state.phone !== null){
+      return <div>Phone: {this.state.phone}</div>;
+    }
+  }
+  renderTwitter = () => {
+    if (this.state.twitter !== null){
+      return <div>Twitter: {this.state.twitter}</div>;
+    }
+  }
+  renderLanguage = () => {
+    let github_language = 'None Listed';
+    if (this.state.github_info.language === null){
+      github_language = 'None Listed';
+    } else {
+      github_language = this.state.github_info.language;
+    }
+    return <div>Github Language: {github_language}</div>;
+  }
+
+
+  renderDashboardContent = () => {
+
+    if(this.hasGH()) {
+      return <React.Fragment>
+      <div>Profile Information:</div>
+        <br/>
+      <div>Email: {this.state.email}</div>
+        {this.renderPhone()}
+        {this.renderTwitter()}
+      <div>Github Page: <a href={"https://github.com/"+this.state.github_info.github_username}>{this.state.github_info.github_username}</a> </div>
+        {this.renderLanguage()}
+      </React.Fragment>;
+    }
+    else {
+      return <React.Fragment>
+      <div>Profile Information:</div>
+        <br/>
+      <div>Email: {this.state.email}</div>
+      {this.renderPhone()}
+      {this.renderTwitter()}
+      </React.Fragment>;
+    }
```

Comment:
React has some inline conditional rendering options: https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator

With this, you could probably write this block as
```
      return <React.Fragment>
      <div>Profile Information:</div>
      <br/>
      <div>Email: {this.state.email}</div>
        {this.renderPhone()}
        {this.renderTwitter()}
        { this.hasGH() && 
            <React.Fragment>
                 <div>Github Page: <a href= 
                 {"https://github.com/"+this.state.github_info.github_username}> 
                 {this.state.github_info.github_username}</a> </div>
                 {this.renderLanguage()}
            </React.Fragment>
       }
      </React.Fragment>;
```